### PR TITLE
turbo: 1.7.0 -> 1.8.1

### PR DESCRIPTION
**Pull request by mausch**
###### Description of changes

Updates to latest release.
Also fixes https://github.com/NixOS/nixpkgs/pull/200788#issuecomment-1438069262
Thanks for the package @dlip and reviewers !

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mausch** [comment]
Hmm some dependency still seems to be missing here...  it crashes when I run `turbo run build` on a project:

```
thread 'main' panicked at 'Failed to execute turbo.: Os { code: 2, kind: NotFound, message: "No such file or directory" }', crates/turborepo/src/main.rs:50:10
stack backtrace:
   0:           0x9cf900 - std::backtrace_rs::backtrace::libunwind::trace::h4a7f1a9dbfd0050f
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:           0x9cf900 - std::backtrace_rs::backtrace::trace_unsynchronized::h5b828858f541f66d
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:           0x9cf900 - std::sys_common::backtrace::_print_fmt::ha0d7f7fa2424466f
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/sys_common/backtrace.rs:65:5
   3:           0x9cf900 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he50c5143f1ab75d4
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/sys_common/backtrace.rs:44:22
   4:           0xa11c5e - core::fmt::write::hc787a3d04103fa2b
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/core/src/fmt/mod.rs:1209:17
   5:           0x9c9095 - std::io::Write::write_fmt::h675091bbbcb60bb9
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/io/mod.rs:1682:15
   6:           0x9cf6c5 - std::sys_common::backtrace::_print::h652b6c4ccf8840dc
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/sys_common/backtrace.rs:47:5
   7:           0x9cf6c5 - std::sys_common::backtrace::print::hed6a8016a9dd162e
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/sys_common/backtrace.rs:34:9
   8:           0x9d10ff - std::panicking::default_hook::{{closure}}::hf463edb68a4f4015
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:267:22
   9:           0x9d0e3a - std::panicking::default_hook::h82bd8191dd7139df
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:286:9
  10:           0x9d17fc - std::panicking::rust_panic_with_hook::he0a74a42970414b0
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:688:13
  11:           0x9d1597 - std::panicking::begin_panic_handler::{{closure}}::h7940a88ccdfc66ac
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:579:13
  12:           0x9cfdac - std::sys_common::backtrace::__rust_end_short_backtrace::h11489a9243f86fef
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/sys_common/backtrace.rs:137:18
  13:           0x9d12b2 - rust_begin_unwind
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:575:5
  14:           0x426733 - core::panicking::panic_fmt::ha40ac504f27f693b
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/core/src/panicking.rs:65:14
  15:           0x4269f3 - core::result::unwrap_failed::h688862f4f03a9f36
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/core/src/result.rs:1791:5
  16:           0x4299ba - turbo::main::h06ba96bf1592f8b5
  17:           0x428753 - std::sys_common::backtrace::__rust_begin_short_backtrace::hdf39de5105f0d646
  18:           0x42806d - std::rt::lang_start::{{closure}}::h341c3aa093144fbe
  19:           0x9c16fe - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h7b5bc0122283ecc9
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/core/src/ops/function.rs:286:13
  20:           0x9c16fe - std::panicking::try::do_call::h5bd05eda45150b51
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:483:40
  21:           0x9c16fe - std::panicking::try::h14eef84f9d16b8c4
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:447:19
  22:           0x9c16fe - std::panic::catch_unwind::h2f8f5a68220e39dc
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panic.rs:137:14
  23:           0x9c16fe - std::rt::lang_start_internal::{{closure}}::h4315d35388392089
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/rt.rs:148:48
  24:           0x9c16fe - std::panicking::try::do_call::h502afca58aff5254
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:483:40
  25:           0x9c16fe - std::panicking::try::h60864fc3302cc007
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panicking.rs:447:19
  26:           0x9c16fe - std::panic::catch_unwind::h39ab24ef2eb872b2
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/panic.rs:137:14
  27:           0x9c16fe - std::rt::lang_start_internal::h9a3de212315a782d
                               at /rustc/215e3cd218b83b8a3152d84d92f17109253c25e1/library/std/src/rt.rs:148:20
  28:           0x429b05 - main
```

Can anyone check if you get the same error?

File: pkgs/tools/misc/turbo/default.nix [hunk 1]
```diff
@@ -64,7 +64,7 @@ rustPlatform.buildRustPackage rec {
   ];
   RELEASE_TURBO_CLI = "true";
 
-  cargoSha256 = "ENw6NU3Fedd+OJEEWgL8A54aowNqjn3iv7rxlr+/4ZE=";
+  cargoSha256 = "sha256-swq8cOAM0AOvFtyHLyJgi/ngZ8vxGxylINMFzJH/Pd8=";
   RUSTC_BOOTSTRAP = 1;
   nativeBuildInputs = [
     pkg-config
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/turbo/default.nix:70]: Please add CoreServices for darwin to fix the darwin build

Comment (mausch) [reply pkgs/tools/misc/turbo/default.nix:73]: No idea about any of that tbh... I grepped nixpkgs and copied something in https://github.com/NixOS/nixpkgs/pull/217476/commits/23f14c7f98f7d74b370f7870a8024eb2f585047d , is that correct?

Comment (dotlambda) [reply pkgs/tools/misc/turbo/default.nix:73]: You need to add
```
inherit (darwin.apple_sdk.frameworks) CoreServices;
```
to `pkgs/top-level/all-packages.nix`.

Comment (mausch) [reply pkgs/tools/misc/turbo/default.nix:73]: done in 854da6f0bc17c78c1b18f1f5763c3acfe787e586

File: pkgs/tools/misc/turbo/default.nix [hunk 2]
```diff
@@ -12,22 +12,23 @@
 , extra-cmake-modules
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/turbo/default.nix:12]:
```

nix-env failed:
error: undefined variable 'stdenv'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-0/pkgs/tools/misc/turbo/default.nix:41:33:

           40|
           41|     buildInputs = lib.optionals stdenv.isDarwin [ CoreServices ];
             |                                 ^
           42|
```

Comment (mausch) [reply pkgs/tools/misc/turbo/default.nix:13]: fixed in https://github.com/NixOS/nixpkgs/pull/217476/commits/5eefe458715e624367ccd2287944a180436376c2

**Comment by mausch** [comment]
```
ld: framework not found CoreServices
```

No idea what this means.
If Darwin doesn't work for whatever reason I could just flag this as Linux-only?
I don't have a Mac so there's not much I can do here really.

File: pkgs/tools/misc/turbo/default.nix [hunk 3]
```diff
@@ -12,22 +13,23 @@
...
 }:
 let
-  version = "1.7.0";
+  version = "1.8.1";
   src = fetchFromGitHub {
     owner = "vercel";
     repo = "turbo";
     rev = "v${version}";
-    sha256 = "YTuEv2S3jNV2o7HJML+P6OMazgwgRhUPnd/zaTWfDWs=";
+    sha256 = "sha256-CkGi7qtFcioQAHeqKkHGrI/iMu8MnlK7IVmCB4AqIfo=";
```
Comment (mweinelt) [inline pkgs/tools/misc/turbo/default.nix:24]:
```suggestion
    hash = "sha256-CkGi7qtFcioQAHeqKkHGrI/iMu8MnlK7IVmCB4AqIfo=";
```

Comment (mausch) [reply pkgs/tools/misc/turbo/default.nix]: Done in e1cce615a65c21adbc2bac3c4be79d2ae7b057b5

**Comment by mweinelt** [comment]
@ofborg build turbo

**Comment by SuperSandro2000** [comment]
Tests are failing

```
--- FAIL: TestWaitForCookie (0.59s)
    cookie_test.go:25: assertion failed: error is not nil: timed out waiting for initial fsevents cookie: filewatching failed to start: Start
--- FAIL: TestWaitForCookieAfterClose (0.54s)
    cookie_test.go:46: assertion failed: error is not nil: timed out waiting for initial fsevents cookie: filewatching failed to start: Start
--- FAIL: TestWaitForCookieTimeout (0.56s)
    cookie_test.go:69: assertion failed: error is not nil: timed out waiting for initial fsevents cookie: filewatching failed to start: Start
--- FAIL: TestWaitForCookieWithError (0.73s)
    cookie_test.go:90: assertion failed: error is not nil: timed out waiting for initial fsevents cookie: filewatching failed to start: Start
--- FAIL: TestFileWatching (1.48s)
    filewatcher_test.go:108: assertion failed: error is not nil: timed out waiting for initial fsevents cookie: filewatching failed to start: fw.Start
FAIL
FAIL    github.com/vercel/turbo/cli/internal/filewatcher        4.166s
FAIL
error: builder for '/nix/store/90ajh0xmsi0s4hkqvr7z52facinqhms4-go-turbo-1.8.1.drv' failed with exit code 1;
```

**Comment by mausch** [comment]
@SuperSandro2000 Can you send me a link to that? 
As far as I can see all checks are currently green except for the Darwin ones.

**Comment by iivvaannxx** [comment]
Hi! I tried your pull request in a custom `flake.nix` file to install turbo in a `devShell`. It builds and works as expected, until I try to run `turbo dev` on a default project created with `pnpx create-turbo@latest`.

```
> turbo run dev

thread ‘main’ panicked at ‘Failed to execute turbo.: Os { code: 2, kind: NotFound, message: “No such file or directory” }’ crates/turborepo/src/main.rs:50:10
```

Do you have any clue and why this could be happening?

**Comment by mausch** [comment]
@iivvaannxx yeah you're likely running `turbo` from node_modules instead of the one from nix.
Run `which turbo` to confirm.
Here's the shellHook I use:
```
        shellHook = ''
          npm i # for some reason yarn crashes without this (?)
          yarn
          rm -rf node_modules/.bin/turbo # use the one from nix
        '';
```

**Comment by dlip** [comment]
Rather than removing the binary, you can set the environment variable:

TURBO_BINARY_PATH="${pkgs.turbo}/bin/turbo"

**Comment by iivvaannxx** [comment]
Thank you both @dlip @mausch! I ended up exporting the environment variable and it worked. However, and I don't know why, if i run `which turbo` and `echo $TURBO_BINARY_PATH` they show exactly the same output, so it seems it wasn't actually using the `node_modules` binary. Even though, exporting the environment variable worked. Now the dev servers can start.

But, I get this rare warning:

```
failed to contact turbod. Continuing in standalone mode: connection to turbo daemon process failed. Please ensure the following:
        - the process identified by the pid in the file at /tmp/nix-shell.uhUxl2/turbod/d85cf2e7a4fada2b/turbod.pid is not running, and remove /tmp/nix-shell.uhUxl2/turbod/d85cf2e7a4fada2b/turbod.pid
        - check the logs at /home/ivan/.local/share/turborepo/logs/d85cf2e7a4fada2b-Portfolio.log
        - the unix domain socket at /tmp/nix-shell.uhUxl2/turbod/d85cf2e7a4fada2b/turbod.sock has been removed
        You can also run without the daemon process by passing --no-daemon
```

Which i am not sure what it means (I'm quite new using `turborepo`). And a little below, on the workspaces logs I get:

```
docs:dev: ready - started server on 0.0.0.0:3001, url: http://localhost:3001
web:dev: ready - started server on 0.0.0.0:3000, url: http://localhost:3000
web:dev: npm ERR! code ENOWORKSPACES
web:dev: npm ERR! This command does not support workspaces.
web:dev: 
web:dev: npm ERR! A complete log of this run can be found in:
web:dev: npm ERR!     /home/ivan/.npm/_logs/2023-03-11T17_42_53_315Z-debug-0.log
docs:dev: npm ERR! code ENOWORKSPACES
docs:dev: npm ERR! This command does not support workspaces.
docs:dev: 
docs:dev: npm ERR! A complete log of this run can be found in:
docs:dev: npm ERR!     /home/ivan/.npm/_logs/2023-03-11T17_42_53_326Z-debug-0.log
```

So it seems that something is not going as it should be. If I can't fix this I can workaround it using Docker to run the development environment but I would like to use my system. Also, if this shouldn't be here, tell me where I should ask it and I will do it straight away.

**Comment by mausch** [comment]
Closing this, superseded by https://github.com/NixOS/nixpkgs/pull/220631
I don't know why I bother wasting my time sending update PRs to nixpkgs, this has happened to me several times already...

**Pull request closed without merging**